awscli/customizations/configservice/subscribe.py

```diff
@@ -138,16 +140,25 @@ def prepare_bucket(self, s3_path):
         bucket_exists = self._check_bucket_exists(bucket)
         if not bucket_exists:
             self._create_bucket(bucket)
+            sys.stdout.write('Using new S3 bucket: %s\n' % bucket)
+        else:
+            sys.stdout.write('Using existing S3 bucket: %s\n' % bucket)
         return bucket, key
 
     def _check_bucket_exists(self, bucket):
         bucket_exists = True
+        self._s3_client.meta.events.unregister(
+            'after-call',
+            unique_id='awscli-error-handler')
         try:
             # See if the bucket exists by running a head bucket
             self._s3_client.head_bucket(Bucket=bucket)
-        except Exception:
-            # If a client error is thrown than the bucket does not exist.
-            bucket_exists = False
+        except ClientError as e:
+            # If a client error is thrown. Check that it was a 404 error.
+            # If it was a 404 error, than the bucket does not exist.
+            error_code = int(e.response['Error']['Code'])
+            if error_code == 404:
+                bucket_exists = False
```

Comment:
@jamesls 
I tested it and I have had confirmation it works. I had to disable the error handler as well:
https://github.com/aws/aws-cli/pull/1223/files#diff-9acfa1e377b87026a3322468d0755a93R150

@danielgtaylor 
As to consolidating the logic, I could but it is not completely the same logic. I actually think it may be a potential bug in the cloudtrail commands as you handle 404 and 403 errors the same when they are not really the same relative to determining if a bucket exists. Is my concern correct?

Let me know.